doc/Language/syntax.pod6

```diff
@@ -223,109 +223,214 @@ say 'code again';
+in ordinary identifiers. Use cases include situations where a set of entities shares
+a common "short" name, but still needs for each of its elements to be identifiable
+individually. For example, you might use a module whose short name is C<Dog>, while
+its long name includes its naming authority and version:
 
-Identifiers can contain colon pairs. The entire colon pair becomes part
-of the name of the identifier.
+=begin code :skip-test
+Dog:auth<Somebody>:ver<1.0>         # Long module names including author and version
+Dog:auth<Somebody>:ver<2.0>
 
-    my $foo::bar = 1;
-    say MY::.keys;
-    # OUTPUT: «($=pod !UNIT_MARKER EXPORT $_ $! ::?PACKAGE
-    #          GLOBALish $¢ $=finish $/ $foo:bar $foo:bar<2>
-    #          $?PACKAGE)␤»
-    say OUR::.key; # OUTPUT: «foo␤»
-
-The last sentence shows how the C<foo> package has been created
-automatically, as a deposit for variables in that namespace.
-
-Identifiers can also contain single colons, and in that case they can
-add I<symbols> within angle brackets. This is a mechanism that supports
-interpolation; which is also supported by colonpairs if the interpolated
-variable is inserted in parentheses. Please note that resolution of
-names often happens at compile time, so interpolation values must be
-known at compile time.
-
-    my $foo:bar<2> = 2;
-    constant $c = 42;
-    my $a:foo<42> = "answer";
-    say $a:foo«$c»;    # OUTPUT: «answer␤»
-    my $buz = "qux";
-    my $bur::quux = 7;
-    say $bur::($buz);  # OUTPUT: «7␤»
+------------------------------------
+
+use Dog:auth<Somebody>:ver<2.0>;    # Selection of second module causes its full name
+                                    # to be aliased to the short name for the rest of
+                                    # the lexical scope, allowing a declaration like
+my Dog $spot .= new("woof");        # this.
+=end code
+
+Similarly, sets of operators work together in various syntactic categories with names
+like C<prefix>, C<infix>, C<postfix>, etc. The long, official names of these operators,
+however, often contain characters that are excluded from ordinary identifiers:
+
+    infix:<+>                       # the official name of the operator in $a + $b
+    infix:<*>                       # the official name of the operator in $a * $b
+    infix:«<=»                      # the official name of the operator in $a <= $b
+
+For all such uses, you can append one or more subscript-like adverbial forms or "colon
+pairs" to an ordinary identifier to create a so-called I<extended identifier>. When
+appended to an identifier (that is, in postfix position), the adverbial syntax
+generates unique variants of that identifier.
+
+The generic colon pair notation acceptable in identifiers is C<:key<value>> : it starts
+with a single colon C<:>, followed by an ordinary identifier C<key>, in turn followed
+by some quoting bracketing construct, such as C«< >», C<« »> or C<[' ']>, which quotes
+one or more arbitrary characters C<value>:
+
```

Comment:
Some examples below don't follow that form, for instance the first one does not have an identifier behind the colon. It's not "some" bracketing constructs, it's precisely those ones.

Reply:
Text revised.